src/main/java/data/Database.java

```diff
@@ -907,6 +908,8 @@ public String getHashedPassword(String username)
 			ResultSet results = pstmt.executeQuery();
 			if (results.next())
 				password = results.getString(1);
+			else
+				return null;
```

Comment:
This can now return null on failure condition, but it doesn't look like the javadocs were updated to reflect this.